cirq/devices/grid_qubit.py

```diff
@@ -38,6 +38,61 @@ def is_adjacent(self, other: ops.Qid) -> bool:
         return (isinstance(other, GridQubit) and
                 abs(self.row - other.row) + abs(self.col - other.col) == 1)
 
+    @staticmethod
+    def square(size: int, top: int = 0, left: int = 0) -> List['GridQubit']:
+        """Returns a square of GridQubits
+
+        Args:
+            size: Length of a side of the square
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+
+        Returns:
+            A list of GridQubits filling in a square grid
+        """
+        return [GridQubit(row, col)
+                for row in range(top, top+size)
+                for col in range(left, left+size)]
+
+    @staticmethod
+    def rect(rows: int, cols: int,
+             top: int = 0, left: int = 0) -> List['GridQubit']:
+        """Returns a rectangle of GridQubits
+
+        Args:
+            rows: Number of rows in the rectangle
+            cols: Number of columns in the rectangle
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+
+        Returns:
+            A list of GridQubits filling in a rectangular grid
+        """
+        return [GridQubit(row, col)
+                for row in range(top, top+rows)
+                for col in range(left, left+cols)]
+
+    @staticmethod
+    def from_pic(s: str) -> List['GridQubit']:
+        """Parse ASCIIart device layout into info about qubits and connectivity.
+
+        Args:
+            s: String representing the qubit layout. Each line represents a row,
```

Comment:
Give two examples of diagrams. Specify that the top row is row 0, left column is column 0.

Reply:
I added two example diagrams, one where all qubits have the same character and one where not at qubits do. I only described the shape of grid in the picture rather than prescribing the exact output list of GridQubits.